teamcity: failed test: TestGORM

The following tests appear to have failed on release-19.1 (Examples-ORMs): TestGORM/FirstRun/GeneratedColumns/ProductsTable, TestGORM/FirstRun/RetrieveFromAPIAfterDependentCreation/Order, TestGORM/FirstRun/GeneratedColumns, TestGORM/FirstRun/RetrieveFromAPIAfterInitialCreation/Customers, TestGORM/FirstRun/RetrieveFromAPIBeforeCreation/Orders, TestGORM/FirstRun/RetrieveFromAPIAfterDependentCreation, TestGORM, TestGORM/FirstRun/GeneratedColumns/OrdersTable, TestGORM/FirstRun/GeneratedColumns/CustomersTable, TestGORM/FirstRun/RetrieveFromAPIBeforeCreation/Customers, TestGORM/FirstRun/EmptyTables/CustomersTable, TestGORM/FirstRun, TestGORM/FirstRun/CreateOrder, TestGORM/FirstRun/RetrieveFromAPIAfterInitialCreation, TestGORM/FirstRun/EmptyTables/ProductsTable, TestGORM/FirstRun/CreateCustomer, TestGORM/FirstRun/EmptyTables, TestGORM/FirstRun/RetrieveFromAPIBeforeCreation/Products, TestGORM/FirstRun/EmptyTables/OrderProductsTable, TestGORM/FirstRun/RetrieveFromAPIBeforeCreation, TestGORM/FirstRun/GeneratedTables, TestGORM/FirstRun/GeneratedColumns/OrderProductsTable, TestGORM/FirstRun/RetrieveFromAPIAfterInitialCreation/Products, TestGORM/SecondRun, TestGORM/FirstRun/EmptyTables/OrdersTable, TestGORM/FirstRun/CreateProduct

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestGORM).

[#1317035](https://teamcity.cockroachdb.com/viewLog.html?buildId=1317035):


```
TestGORM/FirstRun/CreateOrder
--- FAIL: Examples-ORMs/TestGORM/FirstRun/CreateOrder (0.000s)
Test ended in panic.




TestGORM
--- FAIL: Examples-ORMs/TestGORM (0.000s)
Test ended in panic.




TestGORM/FirstRun/CreateCustomer
--- FAIL: Examples-ORMs/TestGORM/FirstRun/CreateCustomer (0.000s)
Test ended in panic.




TestGORM/FirstRun/RetrieveFromAPIBeforeCreation/Orders
--- FAIL: Examples-ORMs/TestGORM/FirstRun/RetrieveFromAPIBeforeCreation/Orders (0.000s)
Test ended in panic.




TestGORM/FirstRun/RetrieveFromAPIBeforeCreation/Customers
--- FAIL: Examples-ORMs/TestGORM/FirstRun/RetrieveFromAPIBeforeCreation/Customers (0.000s)
Test ended in panic.




TestGORM/FirstRun/CreateProduct
--- FAIL: Examples-ORMs/TestGORM/FirstRun/CreateProduct (0.000s)
Test ended in panic.




TestGORM/FirstRun/RetrieveFromAPIAfterInitialCreation/Products
--- FAIL: Examples-ORMs/TestGORM/FirstRun/RetrieveFromAPIAfterInitialCreation/Products (0.000s)
Test ended in panic.




TestGORM/FirstRun/GeneratedColumns
--- FAIL: Examples-ORMs/TestGORM/FirstRun/GeneratedColumns (0.000s)
Test ended in panic.




TestGORM/FirstRun/RetrieveFromAPIAfterDependentCreation/Order
--- FAIL: Examples-ORMs/TestGORM/FirstRun/RetrieveFromAPIAfterDependentCreation/Order (0.000s)
Test ended in panic.




TestGORM/FirstRun/GeneratedColumns/ProductsTable
--- FAIL: Examples-ORMs/TestGORM/FirstRun/GeneratedColumns/ProductsTable (0.000s)
Test ended in panic.




TestGORM/FirstRun/EmptyTables/OrderProductsTable
--- FAIL: Examples-ORMs/TestGORM/FirstRun/EmptyTables/OrderProductsTable (0.000s)
Test ended in panic.




TestGORM/FirstRun/RetrieveFromAPIAfterInitialCreation/Customers
--- FAIL: Examples-ORMs/TestGORM/FirstRun/RetrieveFromAPIAfterInitialCreation/Customers (0.000s)
Test ended in panic.




TestGORM/FirstRun/EmptyTables
--- FAIL: Examples-ORMs/TestGORM/FirstRun/EmptyTables (0.000s)
Test ended in panic.




TestGORM/FirstRun/RetrieveFromAPIBeforeCreation
--- FAIL: Examples-ORMs/TestGORM/FirstRun/RetrieveFromAPIBeforeCreation (0.000s)
Test ended in panic.




TestGORM/FirstRun/GeneratedColumns/OrdersTable
--- FAIL: Examples-ORMs/TestGORM/FirstRun/GeneratedColumns/OrdersTable (0.000s)
Test ended in panic.




TestGORM/FirstRun/GeneratedColumns/CustomersTable
--- FAIL: Examples-ORMs/TestGORM/FirstRun/GeneratedColumns/CustomersTable (0.000s)
Test ended in panic.




TestGORM/FirstRun/EmptyTables/CustomersTable
--- FAIL: Examples-ORMs/TestGORM/FirstRun/EmptyTables/CustomersTable (0.000s)
Test ended in panic.




TestGORM/FirstRun/EmptyTables/ProductsTable
--- FAIL: Examples-ORMs/TestGORM/FirstRun/EmptyTables/ProductsTable (0.000s)
Test ended in panic.




TestGORM/FirstRun/GeneratedColumns/OrderProductsTable
--- FAIL: Examples-ORMs/TestGORM/FirstRun/GeneratedColumns/OrderProductsTable (0.000s)
Test ended in panic.




TestGORM/FirstRun/RetrieveFromAPIBeforeCreation/Products
--- FAIL: Examples-ORMs/TestGORM/FirstRun/RetrieveFromAPIBeforeCreation/Products (0.000s)
Test ended in panic.




TestGORM/FirstRun/RetrieveFromAPIAfterInitialCreation
--- FAIL: Examples-ORMs/TestGORM/FirstRun/RetrieveFromAPIAfterInitialCreation (0.000s)
Test ended in panic.




TestGORM/FirstRun/GeneratedTables
--- FAIL: Examples-ORMs/TestGORM/FirstRun/GeneratedTables (0.000s)
Test ended in panic.




TestGORM/FirstRun/EmptyTables/OrdersTable
--- FAIL: Examples-ORMs/TestGORM/FirstRun/EmptyTables/OrdersTable (0.000s)
Test ended in panic.




TestGORM/SecondRun
--- FAIL: Examples-ORMs/TestGORM/SecondRun (0.000s)
Test ended in panic.

------- Stdout: -------
Makefile:31: recipe for target 'test' failed
Makefile:36: recipe for target 'dockertest' failed



TestGORM/FirstRun
--- FAIL: Examples-ORMs/TestGORM/FirstRun (0.000s)
Test ended in panic.




TestGORM/FirstRun/RetrieveFromAPIAfterDependentCreation
--- FAIL: Examples-ORMs/TestGORM/FirstRun/RetrieveFromAPIAfterDependentCreation (0.000s)
Test ended in panic.





```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/opt/memo/logical_props_builder.go
+++ b/pkg/sql/opt/memo/logical_props_builder.go
@@ -1128,9 +1128,13 @@ func (b *logicalPropsBuilder) buildFiltersItemProps(item *FiltersItem, scalar *p
 	// Constraints
 	// -----------
 	cb := constraintsBuilder{md: b.mem.Metadata(), evalCtx: b.evalCtx}
-	scalar.Constraints, scalar.TightConstraints = cb.buildConstraints(item.Condition)
-	if scalar.Constraints.IsUnconstrained() {
+	// TODO(rytaft): Using local variables here to avoid a data race. It would be
+	// better to avoid lazy building of props altogether.
+	constraints, tightConstraints := cb.buildConstraints(item.Condition)
+	if constraints.IsUnconstrained() {
 		scalar.Constraints, scalar.TightConstraints = nil, false
+	} else {
+		scalar.Constraints, scalar.TightConstraints = constraints, tightConstraints
 	}
 
 	// Functional Dependencies

